libcnb-data/src/launch.rs

```diff
@@ -128,32 +128,36 @@ libcnb_newtype!(
     /// ```
     ProcessType,
     ProcessTypeError,
-    r"^[[:alnum:]\._-]+$"
+    r"^[[:alnum:]._-]+$"
```

Comment:
This change is a no-op - the escaping is unnecessary, as confirmed by the new tests. The new regex style is now consistent with that used by the other types.

Reply:
I was doubtful when I read this comment and checked. You're right, it does not change the validation because of the context where the `.` is used in. However, what do we gain by removing it? I argue that having the escape there makes is more obvious that the regex is correct and does not match any character at this location.

Definitely not a blocker for me, we can move forward as-is, but maybe the above is worth considering.

Reply:
@Malax So I first wondered if the `\` might be taken literally inside the `[]` and actually had to test to confirm it didn't. The other usages don't escape the `.`.

My main preference is to (a) be consistent (so if we want to use this, we should change the others), (b) have tests so we don't have to think about this (which we do, for the `.` case).

Reply:
Again, not a blocker for me :) This is really just about the case if someone reads the regex and wonders. Super minor. Happy to move forward! :)